src/base/preferences.cpp

```diff
@@ -1056,45 +1045,45 @@ bool Preferences::isMagnetLinkAssocSet()
 
 void Preferences::setTorrentFileAssoc(const bool set)
 {
-    QSettings settings("HKEY_CURRENT_USER\\Software\\Classes", QSettings::NativeFormat);
+    QSettings settings(u"HKEY_CURRENT_USER\\Software\\Classes"_qs, QSettings::NativeFormat);
 
     // .Torrent association
     if (set)
     {
-        const QString oldProgId = settings.value(".torrent/Default").toString();
-        if (!oldProgId.isEmpty() && (oldProgId != "qBittorrent"))
-            settings.setValue(".torrent/OpenWithProgids/" + oldProgId, "");
-        settings.setValue(".torrent/Default", "qBittorrent");
+        const QString oldProgId = settings.value(u".torrent/Default"_qs).toString();
+        if (!oldProgId.isEmpty() && (oldProgId != QLatin1String("qBittorrent")))
```

Comment:
```suggestion
        if (!oldProgId.isEmpty() && (oldProgId != u"qBittorrent"))
```

Reply:
I failed to understand which specific overload will be used, can you point out?

Reply:
> I failed to understand which specific comparison overload will be used, can you point out?

It looks like there are gaps in the Qt documentation. Actually it calls `operator!=(const QString &, const char16_t *)` in Qt 6.2.

Reply:
>It looks like there are gaps in the Qt documentation. Actually it calls operator!=(const QString &, const char16_t *) in Qt 6.2.

Weird thing is, it compiles alright with Qt5 but I couldn't found any matching overload in qstring header...

Reply:
> Weird thing is, it compiles alright with Qt5 but I couldn't found any matching overload in qstring header... looks like magic.

The debugger will help you. The functions actually related to QString can be in different files, e.g. QStringView, QStringBuilder etc.